### PR TITLE
Fix docs builds w/o SGX SDK installed

### PR DESCRIPTION
**Pull request by jcape**
# This PR

- Make sure everything is using utilities from `mc-sgx-core-build`.
- Add `mc_sgx_core_build::sgx_include_path()`, use `core/build/headers` if `SGX_SDK` is not set.
- Add headers from the Intel SGX SDK to `core/build/headers/` dir.
- Add a task to build docs to the `build` GHA workflow.

# Future Work

- Handling the location of the headers better (e.g. submodules)

### CODE REVIEW

File: dcap_tvl/sys/build.rs [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 
 fn main() {
-    let sgx_library_path = mc_sgx_core_build::sgx_library_path();
-    rustc_link_lib!("static=sgx_dcap_tvl");
-    rustc_link_search!(&format!("{}/lib64", sgx_library_path));
+    let include_path = mc_sgx_core_build::sgx_include_path();
+    cargo_emit::rerun_if_changed!(include_path);
+
+    let link_path = mc_sgx_core_build::sgx_library_path();
+    cargo_emit::rustc_link_search!(link_path);
```
Comment (nick-mobilecoin) [inline dcap_tvl/sys/build.rs:11]:
```suggestion
    cargo_emit::rustc_link_search!(link_path);
    cargo_emit::rerun_if_changed!(link_path);
```

**Review by nick-mobilecoin — APPROVED** [review]

File: tstdc/sys/build.rs [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
 fn main() {
-    let sgx_library_path = mc_sgx_core_build::sgx_library_path();
-    rustc_link_lib!("static=sgx_tstdc");
-    rustc_link_search!(&format!("{}/lib64", sgx_library_path));
+    let include_path = mc_sgx_core_build::sgx_include_path();
+    cargo_emit::rerun_if_changed!(include_path);
+
+    let link_path = mc_sgx_core_build::sgx_library_path();
+    cargo_emit::rerun_if_changed!(link_path);
+    cargo_emit::rustc_link_search!(&format!("{}", link_path));
```
Comment (nick-mobilecoin) [inline tstdc/sys/build.rs:12]:
```suggestion
    cargo_emit::rustc_link_search!(link_path);
```

**Comment by nick-mobilecoin** [comment]
I think the following should address the test enclave failures, not sure I like the `{}/../bin` but it works for now.
```diff
diff --git a/test_enclave/build.rs b/test_enclave/build.rs
index 0316c8a..0ddcd9e 100644
--- a/test_enclave/build.rs
+++ b/test_enclave/build.rs
@@ -71,7 +71,7 @@ fn build_enclave_definitions<P: AsRef<Path>>(edl_file: P) -> EdgerFiles {
         .expect("Invalid UTF-8 in edl path"));
 
     let sgx_library_path = mc_sgx_core_build::sgx_library_path();
-    let mut command = Command::new(&format!("{}/bin/x64/sgx_edger8r", sgx_library_path));
+    let mut command = Command::new(&format!("{}/../bin/x64/sgx_edger8r", sgx_library_path));
     let out_dir = mc_sgx_core_build::build_output_path();
     command
         .current_dir(&out_dir)
@@ -111,7 +111,7 @@ where
             .expect("Invalid UTF-8 in enclave C file"));
     }
 
-    let sgx_library_path = mc_sgx_core_build::sgx_library_path();
+    let sgx_include_path = mc_sgx_core_build::sgx_include_path();
 
     // This `Build` builds a static library.  If we don't omit the
     // `cargo_metadata` then this static library will be linked into
@@ -122,8 +122,8 @@ where
     // be directly linked in.
     Build::new()
         .files(files)
-        .include(format!("{}/include", sgx_library_path))
-        .include(format!("{}/include/tlibc", sgx_library_path))
+        .include(&sgx_include_path)
+        .include(format!("{}/tlibc", sgx_include_path))
         .cargo_metadata(false)
         .compile("enclave");
 
@@ -163,8 +163,8 @@ fn build_dynamic_enclave_binary<P: AsRef<Path>>(static_enclave: P) -> PathBuf {
                 .expect("Invalid UTF-8 in static enclave path"),
         )
         .args(&["-z", "relro", "-z", "now", "-z", "noexecstack"])
-        .arg(&format!("-L{}/lib64/cve_2020_0551_load", sgx_library_path))
-        .arg(&format!("-L{}/lib64", sgx_library_path))
+        .arg(&format!("-L{}/cve_2020_0551_load", sgx_library_path))
+        .arg(&format!("-L{}", sgx_library_path))
         .arg("--no-undefined")
         .arg("--nostdlib")
         .arg("--start-group")
@@ -207,7 +207,7 @@ fn sign_enclave_binary<P: AsRef<Path>>(unsigned_enclave: P) -> PathBuf {
     let signing_key = get_signing_key();
 
     let sgx_library_path = mc_sgx_core_build::sgx_library_path();
-    let mut command = Command::new(format!("{}/bin/x64/sgx_sign", sgx_library_path));
+    let mut command = Command::new(format!("{}/../bin/x64/sgx_sign", sgx_library_path));
     command
         .arg("sign")
         .arg("-enclave")
@@ -259,11 +259,11 @@ fn get_signing_key() -> PathBuf {
 /// # Returns
 /// The full path to resultant untrusted library.
 fn build_untrusted_library<P: AsRef<Path>>(untrusted_file: P) -> PathBuf {
-    let sgx_library_path = mc_sgx_core_build::sgx_library_path();
+    let sgx_include_path = mc_sgx_core_build::sgx_include_path();
     Build::new()
         .file(untrusted_file)
-        .include(format!("{}/include", sgx_library_path))
-        .include(format!("{}/include/tlibc", sgx_library_path))
+        .include(&sgx_include_path)
+        .include(format!("{}/tlibc", sgx_include_path))
         .compile("untrusted");
 
     let mut untrusted_object = mc_sgx_core_build::build_output_path();
@@ -281,10 +281,10 @@ fn build_untrusted_library<P: AsRef<Path>>(untrusted_file: P) -> PathBuf {
 ///
 /// * `header` - The untrusted header file generated from `edger8r`
 fn build_untrusted_bindings<P: AsRef<Path>>(header: P) {
-    let sgx_library_path = mc_sgx_core_build::sgx_library_path();
+    let sgx_include_path = mc_sgx_core_build::sgx_include_path();
     let bindings = Builder::default()
         .header(header.as_ref().to_str().unwrap())
-        .clang_arg(format!("-I{}/include", sgx_library_path))
+        .clang_arg(format!("-I{}", sgx_include_path))
         .parse_callbacks(Box::new(bindgen::CargoCallbacks))
         .blocklist_type("*")
         // limit to only the functions needed 
```

**Comment by awygle** [comment]
Looks like it's still failing due to missing the edgr8r header.

**Pull request merged**